CMakeLists.txt

```diff
@@ -921,14 +921,14 @@ if(SDL_ASSEMBLY)
       cmake_pop_check_state()
 
       if(CPU_SUPPORTS_LSX AND HAVE_LSXINTRIN_H)
-        list(APPEND EXTRA_CFLAGS "-mlsx")
+        set_property(SOURCE "${SDL2_SOURCE_DIR}/src/video/yuv2rgb/yuv_rgb_lsx.c" APPEND PROPERTY COMPILE_OPTIONS "-mlsx")
```

Comment:
CMake 3.0 does not support the `COMPILE_OPTIONS` property on source files. That's why it is used globally in SDL2. 

https://cmake.org/cmake/help/v3.0/manual/cmake-properties.7.html#properties-on-source-files